includes/admin/class-sensei-onboarding-api.php

```diff
@@ -0,0 +1,224 @@
+<?php
+/**
+ * Onboarding REST API.
+ *
+ * @package Sensei\Onboarding
+ * @since   3.1.0
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+/**
+ * Sensei Onboarding REST API endpoints.
+ *
+ * @package Sensei
+ * @author  Automattic
+ * @since   3.1.0
+ */
+class Sensei_Onboarding_API {
+
+	/**
+	 * Main Onboarding instance.
+	 *
+	 * @var Sensei_Onboarding
+	 */
+	private $onboarding;
+
+	const PURPOSES = [ 'share_knowledge', 'generate_income', 'promote_business', 'provide_certification', 'train_employees', 'other' ];
+
+	/**
+	 * Sensei_Onboarding_API constructor.
+	 *
+	 * @param Sensei_Onboarding $onboarding Onboarding instance.
+	 */
+	public function __construct( $onboarding ) {
+		$this->onboarding = $onboarding;
+	}
+
+	/**
+	 * Register the REST API endpoints for Onboarding.
+	 */
+	public function register() {
+		$common = [
+			'endpoint' => [
+				'permission_callback' => [ $this, 'can_user_access_rest_api' ],
+			],
+			'arg'      => [
+				'validate_callback' => 'rest_validate_request_arg',
+			],
+		];
+
+		$welcome_endpoint = [
+			[
+				'methods'  => 'GET',
+				'callback' => [ $this, 'welcome_get' ],
+			],
+			[
+				'methods'  => 'POST',
+				'callback' => [ $this, 'welcome_submit' ],
+				'args'     => [
+					'usage_tracking' => [
+						'required' => true,
+						'type'     => 'boolean',
+					],
+				],
+			],
+		];
+
+		$purpose_endpoint = [
+			[
+				'methods'  => 'GET',
+				'callback' => [ $this, 'purpose_get' ],
+			],
+			[
+				'methods'  => 'POST',
+				'callback' => [ $this, 'purpose_submit' ],
+				'args'     => [
+					'selected' => [
+						'required' => true,
+						'type'     => 'array',
+						'items'    => [
+							'type' => 'string',
+							'enum' => self::PURPOSES,
+						],
+					],
+					'other'    => [
+						'required' => true,
+						'type'     => 'string',
+					],
+				],
+			],
+		];
+
+		$features_endpoint = [
+			[
+				'methods'  => 'GET',
+				'callback' => [ $this, 'features_get' ],
+			],
+			[
+				'methods'  => 'POST',
+				'callback' => [ $this, 'features_submit' ],
+				'args'     => [
+					'selected' => [
+						'required' => true,
+						'type'     => 'array',
+						'items'    => [
+							'type' => 'string',
+							'enum' => $this->onboarding->plugin_slugs,
+						],
+					],
+				],
+			],
+		];
+
+		Sensei_REST_API_Helper::register_endpoints(
```

Comment:
Hmm. This is an old class that hasn't actually been used for production API endpoints as far as I'm aware. I suspect we might want to ditch the old code here and just use current WordPress REST API practices (which have been developed since then).

@jom what do you think?

Reply:
The `register_endpoints` method is a new helper I added, though I was a bit wary of placing it in `Sensei_REST_API_Helper`. The name fit, but we can move it somewhere new.

Reply:
Yeah, I think we should consider `Sensei_REST_API_Controller`, `Sensei_REST_API_Endpoint_Courses`, `Sensei_REST_API_Endpoint_Modules`, `Sensei_REST_API_Endpoint_Version`, and this class (`Sensei_REST_API_Helper`) deprecated for the purposes of this work.